### ISSUE TITLE
Custom location for new subfolders, phrases, or scripts

### ISSUE THREAD

**Issue by Elliria** [autokey/autokey#676]
### Has this issue already been reported?

- [X] I have searched through the existing issues.

### Is this a question rather than an issue?

- [X] This is not a question.

### What type of issue is this?

Enhancement

### Which Linux distribution did you use?

Any.

### Which AutoKey GUI did you use?

Both

### Which AutoKey version did you use?

Any.

### How did you install AutoKey?

Standard versions were installed from the Linux distribution's repository and the beta was installed using pip.

### Can you briefly describe the issue?

In any version of AutoKey using either front end, you're not able to choose the location for new items when clicking the "New..." button.

### Can the issue be reproduced?

Always

### What are the steps to reproduce the issue?

1. Click the "New..." button in the toolbar.
2. Choose to create a subfolder, phrase, or script.
3. Click the "OK" button.
4. Look for your new subfolder, phrase, or script.

### What should have happened?

We should be given a choice of location for the new item since it may not belong in the current location.

### What actually happened?

_The behavior described below is different in each front end and this has been reported in the #675 issue:_

In the Qt front end, the new subfolder, phrase, or script will be placed in the currently-selected folder.

In the GTK front end, the new subfolder, phrase, or script will be placed in the currently-selected folder or in the folder that the currently-selected phrase or script is in.

### Anything else?

This is not an issue if you want to create the new item in your current location, but if, for example, you have the phrase folder selected when you create a new script, it will be placed into the phrase folder since that's where you are.

Although being aware of this is helpful for finding newly-added items that may not be where we expect them to be, it might be a good idea for AutoKey to make the location a choice.

**Comment by josephj11** [comment]
Although the desired behavior is an enhancement, the current behavior seems like  bug to me because the two front ends behave differently.

This is really two separate issues - inconsistency and enhancement, but I don't think it's worthwhile to split them.

Personally, I would prefer the behavior of New working by default in the current open/selected folder to save me a few clicks, but it's not a big deal either way because new action creation is done infrequently compared to most other functions. Ideally, I think it should work the same way New does in a file manager such as Dolphin or Nautilus.

**Comment by Elliria** [comment]
I agree that the current inconsistent behavior is a bug and have now added a note above linking to the separate issue that I had created for it. I'd say that other than the inconsistency that's addressed in issue #675, it does currently behave the way that Dolphin or Nautilus does, since it creates the new item in the current location. If that's the desired behavior, then this issue can be closed.

**Comment by josephj11** [comment]
That's up to you. I voiced my opinion, but that's just one "vote". Generally, "my way" tends to diverge from what most people prefer. That's a big part of why I like Linux. I can have my way and still be in the fold.

Unrelated: FYI Up in the title, it says "2 tasks done". I searched and it turns out that GitHub has a relatively new task list feature. It gets triggered by the use of check boxes like those in your yaml form. I don't believe there's any way to disable it ATM.

**Comment by Elliria** [comment]
> That's up to you. I voiced my opinion, but that's just one "vote". Generally, "my way" tends to diverge from what most people prefer. That's a big part of why I like Linux. I can have my way and still be in the fold.

That's as it should be and it's wonderful when we can each choose our own way. I guess what this issue is about is that there isn't a choice being offered. If having new items created in the current location is what a user wants, then they'll be happy with it as is. If you'd rather be offered a choice of a different location, then you can't currently have that. You must manually move an item after it has been created if you want it to go elsewhere. Those are just two of the possible work-flows people might have and there may be others. I'm big on offering the user choices, which AutoKey doesn't currently do in this regard.
 
> Unrelated: FYI Up in the title, it says "2 tasks done". I searched and it turns out that GitHub has a relatively new task list feature. It gets triggered by the use of check boxes like those in your yaml form. I don't believe there's any way to disable it ATM.

Yeah, I saw that and wondered what it was about. It makes sense, but hopefully they'll let us disable it.

**Comment by josephj11** [comment]
One thing I don't like, but don't know how to fix is that AFAIK, there is nothing to prevent the user from creating scripts and phrases in the same subfolder. But, if a user was thinking in terms of applications, they might want all the scripts and phrases for an application/function/activity in the same subfolder, so, for them, this might be a feature rather than a shortcoming.

Comment copied from #675. Something else to think about when addressing this issue.

**Comment by Elliria** [comment]
It may or may not be possible to get AutoKey to auto-detect whether something is a phrase or script and get it to either prompt you with an "Are you sure?" message or automatically put things where you've specified that you would like them, but that may take some serious rewriting of the code and may not be worth doing. After all, as you pointed out, there are folks who may not want them separated in that way.

**Comment by josephj11** [comment]
You are selecting create new phrase or create new script, so there's no detecting involved. If someone deliberately circumvents that, then it's on them.

**Comment by Elliria** [comment]
You're right and I shouldn't have missed that. I must have either been tired or I didn't have my thinking cap on.
